pkg/operator/ceph/csi/spec.go

```diff
@@ -662,3 +688,265 @@ func (r *ReconcileCSI) validateCSIVersion(ownerInfo *k8sutil.OwnerInfo) (*CephCS
 	}
 	return version, nil
 }
+
+type migrationJobReport struct {
+	nodeName string
+	err      error
+}
+
+func (r *ReconcileCSI) setupCSINetwork() error {
```

Comment:
In the past, Rook tried using daemonsets for some components and we ended up getting rid of them because the operator didn't ultimately have enough control over the pods. I wonder if we are going to hit the same thing here where managing the daemonsets just doesn't match the pattern that will be best for the operator to manage individual nodes. What about the following approach instead?
- For each node that matches affinity/tolerations for the csi driver settings
   - Start a goroutine to initialize this node 
   - Create a deployment for that node (replica 1 to start a single pod, similar to the OSDs)
   - Wait for the pod to start
   - Compute the node's needed multus config
   - Run the job that updates the multus config on the node

If new nodes are added to the cluster, the same procedure is applied just to that node. If the operator detects a single node goes down or has an issue, it can just focus on repairing that node.

With this approach, all nodes can be configured in parallel, failure of a set of nodes would not affect all nodes,  config doesn't need to be rolled back for all nodes after a single node failure, and the operator will have full control for each node.

On the other hand, the same approach should really be possible with the daemonset. Perhaps it would be simpler with the daemonset. 
- Start the daemonset
- For each node where a pod from the daemonset is starting:
   - Start a goroutine for this node 
   - Wait for the pod to start
   - Compute the node's needed multus config
   - Run the job that updates the multus config on the node

Watching for a new node would just have to watch for a new pod that is starting on a node that hasn't initialized yet. 

The point is that all nodes are independent, and the operator can respond quickly to individual node events for starting or being restarted.